src/storage/mvcc/reader/scanner/forward.rs

```diff
@@ -239,6 +246,13 @@ impl<S: Snapshot, P: ScanPolicy<S>> ForwardScanner<S, P> {
             };
 
             if has_lock {
+                if self.cfg.check_can_be_cached && self.can_be_cached {
+                    let lock_value = self.cursors.lock.value(&mut self.statistics.lock);
+                    let lock = Lock::parse(lock_value)?;
```

Comment:
Lock is parsed twice.. @sticnarf Do you have ideas to reduce the extra parsing?

Reply:
I don't have a good idea... But instead of pass `lock_value` in, I prefer to pass `can_be_cached` in...